sdk/servicebus/azure-servicebus/api_samples.py

```diff
@@ -0,0 +1,219 @@
+## Sample Code:
+
+### creation of different entity clients
+
+# Top Level ServiceBusClient
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+sb_client = ServiceBusClient(
+    fully_qualified_namespace="fully_qualified_namespace",
+    crendetial=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+queue_sender = sb_client.get_queue_sender(queue_name="queue_name")
+queue_receiver = sb_client.get_queue_receiver(queue_name="queue_name")
+topic_sender = sb_client.get_topic_sender(topic_name="topic_name")
+subscription_receiver = sb_client.get_subscription_receiver(
+    topic_name="topic_name",
+    subscription_name="subscription_name"
+)
+
+# Queue Sender
+queue_sender = ServiceBusSender.from_connection_string(
+    conn_str="conn_str",
+    queue_name="queue_name"
+)
+
+queue_sender = ServiceBusSender(
+    fully_qualified_namespace="fully_qualified_namespace",
```

Comment:
Is the "non-factory" method (looksl ike TokenCredential) intended to be the encouraged case? Ie, is that why we've made connection string a factory method?